src/full/Agda/TypeChecking/Serialise.hs

```diff
@@ -207,24 +207,30 @@ decode s = do
   -- The decoder is (intended to be) strict enough to ensure that all
   -- such errors can be caught by the handler here.
 
-  (mf, r) <- liftIO $ E.handle (\(E.ErrorCall s) -> noResult s) $ do
-
+  res <- liftIO $ E.handle (\(E.ErrorCall s) -> pure Nothing) $ do
```

Comment:
`s` is dropped here, needlessly it seems.
It seems like we could communicate the reason `s` when using `Left/Right` instead of `Nothing/Just` here.

Reply:
I added `s` propagation and printing in the failure case just below. The original error printing was commented out to not pull `Pretty` dependency, but now the message is just a `String`.